### DISCUSSION TITLE
Programatically Interaction - E2E And Product Tours

### DISCUSSION THREAD

**Discussion by guifeliper** [shadcn-ui/ui#3354] [Q&A]
I am using Driver JS to create a Guide tour for users in my application, some of the items I click programmatically for the user, so for example if there is a button I do a vanilla `.click()` in the element. However, it is not working on Radix/shadcn-ui and I sincerely don't understand why. 

Webpage: https://ui.shadcn.com/docs/components/dropdown-menu
Example using id: 

```
const element = document.querySelector("#radix-\\:r4q\\:");
element.click()
```

Example using XPath: 

```
const xpath = '//*[@id="radix-:r4q:"]';
const result = document.evaluate(xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null);
const element = result.singleNodeValue;

if (element) {
  const clickEvent = new Event('click', { bubbles: true });
  element.dispatchEvent(clickEvent);
}
```

Do you know any thing about this situation?

**Comment by shrix1** [comment]
Are you using ReactJS/NextJS or vanilla JS. 
- if ReactJS/NextJS means try : https://www.npmjs.com/package/reactour

**Reply by guifeliper** [reply]
I'm going to have a look at it. Why do you think they are better than DriverJS?

**Reply by shrix1** [reply]
ReactTour is easy to implemented on React with nice DX + more downloads and nice community around it

**Comment by guifeliper** [comment]
After coming back to this issue, I took a look at their Radix Primitive code and noticed they use onMouseDown, onPointerDown and others to update the value, so instead of setting a click in your Testing library or your programmatically click, you should trigger an event. From my current research, I noticed they are using OnKeyDown on the components that I searched for, so I am considering they have it for all of them, so I suggest the following code. 

    const myDropdown = document.querySelector("#dropdown")
    const keyDownEvent = new KeyboardEvent("keydown", {
          bubbles: true,
          cancelable: true,
          key: "Enter",
          keyCode: 13
        });
    myDropdown.dispatchEvent(keyDown)


However, I don't consider the proper solution, I also don't know why they made that choice of _"disabling"_ the onClick. which, I would love to know. 

You can learn more about their implementation here: https://github.com/radix-ui/primitives/blob/b32a93318cdfce383c2eec095710d35ffbd33a1c/packages/react/tabs/src/Tabs.tsx
